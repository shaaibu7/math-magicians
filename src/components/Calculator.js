import React from 'react';

function Calculator() {
  return (
    <div className="calculator">
      <div className="input">
        <p className="data">0</p>
      </div>

      <div className="keys">
        <button type="button" className="key">AC</button>
        <button type="button" className="key">+/-</button>
        <button type="button" className="key">%</button>
        <button type="button" className="calculate">+</button>
      </div>

      <div className="keys">
        <button type="button" className="key">7</button>
        <button type="button" className="key">8</button>
        <button type="button" className="key">9</button>
        <button type="button" className="calculate">%</button>
      </div>

      <div className="keys">
        <button type="button" className="key">4</button>
        <button type="button" className="key">5</button>
        <button type="button" className="key">6</button>
        <button type="button" className="calculate">-</button>
      </div>

      <div className="keys">
        <button type="button" className="key">1</button>
        <button type="button" className="key">2</button>
        <button type="button" className="key">3</button>
        <button type="button" className="calculate">+</button>
      </div>

      <div className="keys">
        <button type="button" className="key zero">0</button>
        <button type="button" className="key">.</button>
        <button type="button" className="calculate">=</button>
      </div>
    </div>
  );
}

export default Calculator;
