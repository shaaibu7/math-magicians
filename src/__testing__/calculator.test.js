import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Calculator from '../components/Calculator';
import '@testing-library/jest-dom';

test('testing for snapshot of calculator component', () => {
  const calculator = renderer.create(<Calculator />).toJSON();
  expect(calculator).toMatchSnapshot();
});

test('render the AC button in the UI', () => {
  render(<Calculator />);
  const btnAc = screen.getByText(/AC/i);
  expect(btnAc).toBeInTheDocument();
});

test('render the + button in the UI', () => {
  render(<Calculator />);
  const btnPlus = screen.getByText('+');
  expect(btnPlus).toBeInTheDocument();
});

test('render the - button in the UI', () => {
  render(<Calculator />);
  const btnMinus = screen.getByText('-');
  expect(btnMinus).toBeInTheDocument();
});

test('render the % button in the UI', () => {
  render(<Calculator />);
  const btnPercentage = screen.getByText('%');
  expect(btnPercentage).toBeInTheDocument();
});

test('render the รท button in the UI', () => {
  render(<Calculator />);
  const btnDivide = screen.getByText('รท');
  expect(btnDivide).toBeInTheDocument();
});

test('render the = button in the UI', () => {
  render(<Calculator />);
  const btnEqualTo = screen.getByText('=');
  expect(btnEqualTo).toBeInTheDocument();
});
