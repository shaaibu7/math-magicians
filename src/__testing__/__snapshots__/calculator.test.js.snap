// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing for snapshot of calculator component 1`] = `
<div
  className="main"
>
  <h2
    className="head"
  >
    Let\`s do some maths!
  </h2>
  <div
    className="calculator"
  >
    <div
      className="input"
    >
      <input
        className="input data"
        type="text"
        value="0"
      />
    </div>
    <div
      className="keys"
    >
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="calculate"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="keys"
    >
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="calculate"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="keys"
    >
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="calculate"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="keys"
    >
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="calculate"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="keys"
    >
      <button
        className="key zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="key"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="calculate"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
